### ISSUE TITLE
Exif "Orientation" vs "Camera Orientation"

### ISSUE THREAD

**Issue by janaz** [lovell/sharp#547]
Hi Lovell,

I came across an [image](https://i3.au.reastatic.net/original/1492b56e44cfe07b56931e784517b523405cbadfb6bffee285746d3c644559df/image11.jpg) that has both `Orientation` and `Camera Orientation` exif tags set.

The `sharp.rotate()` operation is only inspecting the `Orientation` attribute ignoring the existence of the `Camera Orientation`. As the result the image returned by sharp is a rotated version (270 CW) of the expected vertical image.

```bash
$ exiftool /tmp/image11.jpg  | grep -n Orientation
Orientation                     : Rotate 270 CW
Camera Orientation              : Rotate 270 CW

$ exiftool -n /tmp/image11.jpg  | grep Orientation
Orientation                     : 8
Camera Orientation              : 2
```

I haven't done much analysis, but it looks like `rotate()` operation should be looking at the "difference" between `Orientation` and `Camera Orientation` and apply the rotation that is the result of that difference. In the above example the computation will look like "(Rotate 270 CW) - (Rotate 270 CW) -> (Rotate 0)"

Can the "Camera Orientation" value be exposed in the metadata object?

**Comment by lovell** [comment]
Thanks Tomasz, I wasn't aware of this, we should probably make sharp use the difference between these also.

To workaround this for now, the raw EXIF data returned by the existing `metadata()` operation can be parsed by the `exif-reader` module, e.g. https://github.com/lovell/sharp/blob/master/test/unit/metadata.js#L46-L49

**Comment by puzrin** [comment]
Hm... in Chrome & FF i see image with wrong rotation. Not sure they process cam orientation. Is it something new and not propagated to libraries?

PS. I didn't know about camera orientation, thanks for info.

**Comment by janaz** [comment]
Unfortunately there's not much online resources about the Camera Orientation tag. Apparently it's [specific only to Canon cameras](http://apdev.blogspot.com.au/2013/09/camera-orientation-vs-orientation-exif.html). The EXIF data from the example discussed in the blog doesn't match the EXIF data from my image.

After looking closer image I linked, I start to think that it was processed by software that didn't handle the exif data properly:
* The image is resized to 1024x1536 (proof that it was processed). 
* I wasn't able to find original Canon images with the same combination of "Orientation" and "Camera Orientation" values. The examples I found were: Orientation: Normal, Camera Orientation: 270 CW.

I'm inclined to closing this PR as the issue I reported is most likely a result of incorrect input data. I will reopen when I find some more examples.

@puzrin I also noticed that FF and Chrome render the image in that way. (It's interesting that the same browsers completely ignore the Orientation exif data for <img> elements)

@lovell I didn't know about the `exif-reader`. I'll have a look at this.
